src/v/rpc/simple_protocol.cc

```diff
@@ -103,6 +103,17 @@ simple_protocol::dispatch_method_once(header h, net::server::resources rs) {
       = ssx::spawn_with_gate_then(
           rs.conn_gate(),
           [this, method_id, rs, ctx]() mutable {
+              if (ctx->get_header().version > transport_version::v0) {
+                  netbuf reply_buf;
+                  reply_buf.set_status(rpc::status::version_not_supported);
+                  reply_buf.set_version(transport_version::max_supported);
```

Comment:
What are clients expected to do with this version? Is this more informational than functional?

Reply:
at the moment it is only informational. in reality it probably isn't important if we assume that the backwards looking range of versions that we might incur in any situation is very small (and this should be true for changes to the low-level protocol). but it seems like an easy piece of information to return just in case it is useful in the future.

Reply:
I added a comment here in the code to explain this point.